test/gtest/ucs/test_profile.cc

```diff
@@ -165,24 +165,35 @@ void test_profile::run_profiled_code(int num_iters)
     if (num_threads() == 1) {
         profile_thread_func(&param);
     } else {
-        std::queue<pthread_t> threads;
+        std::vector<pthread_t> threads;
 
         for (int i = 0; i < num_threads(); ++i) {
             pthread_t profile_thread;
             ret = pthread_create(&profile_thread, NULL, profile_thread_func,
                                  (void*)&param);
-            ASSERT_EQ(0, ret);
+            if (ret < 0) {
+                UCS_TEST_MESSAGE << "pthread_create failed:" << strerror(errno);
+                goto out;
+            }
 
-            threads.push(profile_thread);
+            threads.push_back(profile_thread);
         }
 
+out:
         while (!threads.empty()) {
             void *result;
-            ret = pthread_join(threads.front(), &result);
-            ASSERT_EQ(0, ret);
-            threads.pop();
+            ret = pthread_join(threads.back(), &result);
+            if (ret < 0) {
+                UCS_TEST_MESSAGE << "pthread_join failed:" << strerror(errno);
+            }
+
+            threads.pop_back();
         }
     }
+
+    if (ret < 0) {
```

Comment:
what's the point, ret is overridden by pthread_join?

Reply:
ah yes, it can be 0

is it ok to add `error` variable that will be set to 1 if at least one `ret < 0` condition is taken?
and then we will check it here

Reply:
use EXPECT_EQ or ADD_FAILURE to make the test fail later

Reply:
`ADD_FAILURE` is good here, because need to do `goto out`
thank you for the advice!